src/zenml/cli/stack.py

```diff
@@ -230,13 +230,22 @@ def register_stack(
 
         for c_type, name in type_component_mapping.items():
             if name:
-                stack_components[c_type] = [
-                    cli_utils.get_component_by_id_or_name_or_prefix(
-                        client=client,
-                        component_type=c_type,
-                        id_or_name_or_prefix=name,
-                    ).id
-                ]
+                component = cli_utils.get_component_by_id_or_name_or_prefix(
+                    client=client,
+                    component_type=c_type,
+                    id_or_name_or_prefix=name,
+                )
+                stack_components[c_type] = [component.id]
+                if share:
+                    if not component.is_shared:
+                        cli_utils.error(
+                            "You attempted to create a shared stack containing "
+                            f"a private component {c_type}: {name}. This "
+                            f"is not supported. Set the {c_type} to "
+                            f"shared like this and then try re-sharing your "
```

Comment:
same as below, you shouldn't reshare your stack but rather run the register command again I guess